app/controllers/application_controller.rb

```diff
@@ -55,4 +56,30 @@ def permitted_org_bike_search_params
     @stolenness ||= params["stolenness"].present? ? params["stolenness"] : "all"
     params.permit(*Bike.permitted_search_params).merge(stolenness: @stolenness)
   end
+
+  def locale_from_request_header
+    request.env.fetch("HTTP_ACCEPT_LANGUAGE", "").scan(/^[a-z]{2}/).first
+  end
+
+  def locale_from_request_params
+    requested_locale = params.fetch(:locale, "").strip.to_sym
+    requested_locale if I18n.available_locales.include?(requested_locale)
+  end
+
+  def set_locale
+    return unless Flipper.enabled?(:localization, current_user)
+
+    logger.debug("* Params: '#{locale_from_request_params}'")
```

Comment:
Do we want to leave these logger statements in?

Reply:
Not permanently -- I'll add a comment to remove them with the feature flag.